apps/server/src/controllers/customerControllers/authCustomers.ts

```diff
@@ -129,3 +130,108 @@ export const login = async (req: Request, res: Response) => {
       .json({ message: "Internal Server Error" });
   }
 };
+
+export const customerForgotPassword = async (req: Request, res: Response) => {
+  try {
+    let { email } = req.body;
+
+    email = email.trim().toLowerCase();
+
+    const user = await Customers.findOne({ where: { email } });
+
+    if (user) {
+      const longString = generateLongString(80);
+
+      user.resetToken = longString;
+      user.resetTokenExpiry = new Date(Date.now() + 1000 * 60 * 60 * 24);
+      await user.save();
+
+      const transporter = nodemailer.createTransport({
+        service: "gmail",
+        auth: {
+          user: ENV.GMAIL_USER,
+          pass: ENV.GMAIL_PASSWORD,
+        },
+      });
+
+      const mailOptions = {
+        from: ENV.GMAIL_USER,
+        to: email,
+        subject: "Reset your password",
+        text: `Hi, ${user.firstName} ${user.lastName} \n\nPlease use the following link to reset your password \n\n  ${ENV.FE_BASE_URL}/auth/resetPassword?token=${longString}`,
+      };
+
+      console.log(mailOptions.text)
+
+      transporter.sendMail(mailOptions, (err: any, info: { response: string; }) => {
+        if (err) {
+          console.log(err);
+        } else {
+          console.log("Email sent: " + info.response);
+        }
+      });
+    }
+
+    return res.status(HTTP_STATUS_CODE.SUCCESS).json({
+      message: `Password reset link will be sent to your email if you have an account with us.`,
+    });
+  } catch (error) {
+    console.log(error);
+
+    return res.status(HTTP_STATUS_CODE.INTERNAL_SERVER).json({
+      message: [
+        { message: `This is our fault, our team are working to resolve this.` },
+      ],
+    });
+  }
+};
+
+export const customerResetPassword = async (req: Request, res: Response) => {
+  try {
+    const { newPassword, confirmPassword } = req.body;
```

Comment:
Only the newPassword is needed. The frontend will the check of incorrect password.

Reply:
removed